### ISSUE TITLE
Speed up GitHub Windows build

### ISSUE THREAD

**Issue by ghutchis** [OpenChemistry/avogadrolibs#799]
Right now, the GitHub builds for Windows are slow:
- [x] it doesn't cache the third party or downloaded dependencies
- [x] it doesn't use `ninja` so builds only use one (of two) cores

The GitHub action is at https://github.com/OpenChemistry/avogadrolibs/blob/master/.github/workflows/build_cmake.yml

If someone can work with Windows builds, I can help translate commands / scripts into the GitHub action.

**Comment by ghutchis** [comment]
I've also been experimenting with `sccache` and it doesn't seem to work, so suggestions are welcome.

**Comment by adityaomar3** [comment]
Correct me if I am wrong as I can see for windows https://github.com/OpenChemistry/avogadrolibs/blob/8970c8b3695aeff5ba9322208e89845912736748/.github/workflows/build_cmake.yml#L55C13-L55C13
DOPENSSL is used but it supports GNINJA
and as for sccache we can try LLVM Compiler Cache (LCC) it is said to faster than sccache as its algorithms to build are better and different

**Comment by ghutchis** [comment]
I have had trouble using `-G Ninja` on Windows. If you want to try tweaking that build script and can get it to work, that would be *great*

Also,
```
    - name: Grab cache files
      uses: actions/cache@v3
      if: runner.os != 'Windows'
      with:
        path: |
          ${{ runner.workspace }}/build/Downloads
        key: ${{ matrix.config.name }}-downloads
```

This caches the third-party downloads, which also does not seem to work on Windows. (Or, more accurately, the build seems to break when downloads are cached.)

If the LLVM compiler cache works on Ubuntu, Mac, and Windows and can be used instead of `sccache` I would happily switch.

Basically, anything that can speed the builds, particular on Windows would be greatly appreciated!

**Comment by ghutchis** [comment]
You don't have to apologize for asking questions, but it's probably easier if you post in the forum: https://discuss.avogadro.cc

And it would help if you can post as much of the error and what you've done to know how to help, thanks.

**Comment by adityaomar3** [comment]
Thanks, it can be found here https://discuss.avogadro.cc/t/error-while-cloning-git-of-open-chemistry/4830?u=aditya_omar

**Comment by matterhorn103** [comment]
With all the changes to builds, is this still open?

**Comment by ghutchis** [comment]
This is now mostly fixed by #2518 - which uses Ninja parallel builds and caches downloads.
